File: src/aws/lambdas/incremental_distribution/src/main/java/uk/nhs/nhsx/virology/tokengen/CtaProcessorResult.java
package uk.nhs.nhsx.virology.tokengen;

import java.util.Map;

public abstract class CtaProcessorResult {

    public abstract Map<String, String> toResponse();

    public static class Success extends CtaProcessorResult {
        public final String zipFilename;
        private final String message;

        public Success(String zipFilename, String message) {
            this.zipFilename = zipFilename;
            this.message = message;
        }

        @Override
        public Map<String, String> toResponse() {
            return Map.of(
                "result", "success",
                "message", message,
                "filename", zipFilename
            );
        }
    }

    public static class Error extends CtaProcessorResult {
        public final String message;

        public Error(String message) {
            this.message = message;
        }

        @Override
        public Map<String, String> toResponse() {
            return Map.of(
                "result", "error",
                "message", message
            );
        }
    }
}

File: src/aws/lambdas/incremental_distribution/cta/src/main/java/uk/nhs/nhsx/virology/tokengen/CtaProcessorResult.kt
package uk.nhs.nhsx.virology.tokengen

abstract class CtaProcessorResult {
    abstract fun toResponse(): Map<String, String>

    class Success(val zipFilename: String, private val message: String) : CtaProcessorResult() {
        override fun toResponse() = mapOf(
            "result" to "success",
            "message" to message,
            "filename" to zipFilename
        )
    }

    class Error(val message: String) : CtaProcessorResult() {
        override fun toResponse() = mapOf(
            "result" to "error",
            "message" to message
        )
    }
}
